native/src/components/LocationButton.tsx

```diff
@@ -0,0 +1,58 @@
+import LocationIcon from '../assets/IconPlaceholder.svg'
+import SimpleImage from './SimpleImage'
+import styled from 'styled-components/native'
+import { TouchableOpacity, View } from 'react-native'
+import { checkLocationPermission, requestLocationPermission } from '../utils/LocationPermissionManager'
+import React, { ReactElement, useCallback, useEffect } from 'react'
+import { RESULTS } from 'react-native-permissions'
+
+const ICON_SIZE = 45
+
+const Circle = styled(View)`
```

Comment:
I think instead we could just reuse the current button from `CitySelector`, I think that icon is more intuitive and I think it makes sense to use the same design/look in both places.

Reply:
Actually I only copy pasted this button 🙈

Reply:
you mean the location marker? Currently there is only the refresh icon for the location in the `CitySelector`

Reply:
> > I think the behavior of this is a little bit off if the permission is granted but there is no current user location available. If I press the button, the map is just zoomed into to some random place. From other maps I am used to nothing happening instead of displaying a hint/toast if no location is available. Maybe we could use the snackbar for this?
> 
> Did you test this on a real device? I had the same issue when I did not set the Location on the emulator.

Yes, my phone. But it is sometimes a little bit weird as I have no google services. What happens for you if you just disable location on your phone (i.e. not the permission but the setting).

Reply:
Ups, you are right. Then it probably does not make sense...

Reply:
i also suggested to check for standard button or pass the custom icon convenient to the api reference:
https://github.com/react-native-mapbox-gl/maps/blob/master/docs/UserLocation.md
this was just a reminder :)

Reply:
I checked this one. This will replace the location indicator and not a location button. Same goes for the on press, which is triggered when tapping the userlocation

Reply:
I used the `FloatingActionButton` which @maxammann  recommended.